src/zenml/integrations/aws/flavors/sagemaker_orchestrator_flavor.py

```diff
@@ -80,13 +82,17 @@ class SagemakerOrchestratorSettings(BaseSettings):
 
     processor_args: Dict[str, Any] = {}
     input_data_s3_mode: str = "File"
-    input_data_s3_uri: Optional[Union[str, Dict[str, str]]] = None
+    input_data_s3_uri: Optional[Union[str, Dict[str, str]]] = Field(
+        default=None, union_mode="left_to_right"
```

Comment:
Not sure about union mode here: if I pass `{"a":"b"}` it would never resolve as a string, or? Is it mandatory to have `union_mode` at all?

Reply:
The `smart_union` configuration parameter is now removed. Now, the default behavior is `smart_union`, which means if we had a `Union` field that was **not parameterized with `smart_union` before**, we have to manually set `union_mode='left_to_right` to keep the same behavior.

For this specific example, I am not sure if this would have created a problem. But we had other instances where switching from `left_to_right` to `smart_union` broke the logic. Since this `Union` pattern is very widely used in our codebase, I did not have enough time to go through them one-by-one and change them on a case-basis, so I parameterized them all the keep the same behaviour.